packages/server-wallet/src/metrics.ts

```diff
@@ -1,4 +1,33 @@
+import {performance, PerformanceObserver, PerformanceEntry} from 'perf_hooks';
+import fs from 'fs';
+
 import walletConfig from './config';
+import knex from './db/connection';
+
+if (walletConfig.timingMetrics) {
+  if (walletConfig.metricsOutputFile) {
+    fs.writeFileSync(walletConfig.metricsOutputFile, '', {flag: 'w'});
+  }
+  const log = (entry: PerformanceEntry): void => {
+    if (walletConfig.metricsOutputFile) {
+      fs.appendFileSync(walletConfig.metricsOutputFile, JSON.stringify(entry));
+      fs.appendFileSync(walletConfig.metricsOutputFile, '\n');
```

Comment:
I don't think we want to block on the IO here.
```suggestion
      fs.appendFile(walletConfig.metricsOutputFile, JSON.stringify(entry) + '\n', console.error);
```

Reply:
I think we want to block to prevent multiple writes to the file occurring at the same time?

Reply:
hmm... I am definitely wary of synchronous IO. does `fs.appendFile` suffer from that kind of race condition?

While searching around for some literature, I came across: https://stackoverflow.com/questions/3459476/how-to-append-to-a-file-in-node/43370201#43370201

Reply:
@kerzhner might have some clarifying thoughts here?

Reply:
For what it's worth I originally used a `writeStream` but I found mangled output in the log file. I assumed this was a concurrency issue which seemed to be resolved by using the `appendFileSync`, however I didn't dig too deep into it.

Reply:
well, if it's not turned on in production, it doesn't matter _that_ much...